finagle-core/src/test/scala/com/twitter/finagle/InetResolverTest.scala

```diff
@@ -76,4 +78,36 @@ class InetResolverTest extends FunSuite {
     assert(statsReceiver.stat("lookup_ms")().size > 0)
   }
 
+  test("updates using custom future pool") {
+    class TestPool(latch: CountDownLatch) extends FuturePool {
+      private val delegate = FuturePool.unboundedPool
+      override def apply[T](f: => T): Future[T] = {
+        latch.countDown()
+        delegate(f)
+      }
+    }
+
+    val latch = new CountDownLatch(1) // DnsResolver will use the pool
+    val resolvePool = new TestPool(latch)
+
+    val dnsResolverWithPool = new DnsResolver(DefaultStatsReceiver, resolvePool)
+
+    def resolveLoopback(host: String): Future[Seq[InetAddress]] = {
+      if (host.equals("127.0.0.1")) dnsResolverWithPool(host)
+      else Future.exception(new UnknownHostException())
+    }
+
+    val pollInterval = Duration(100, TimeUnit.MILLISECONDS)
```

Comment:
Yes I have seen it, good idea! However, with the current test setup, advancing time isn't really required as `inetResolverWithPool.bind` will count down the latch one time immediately. In other words `pollInterval` isn't really used in the test. 

What I tried to do, is to test this [section](https://github.com/twitter/finagle/blob/develop/finagle-core/src/main/scala/com/twitter/finagle/InetResolver.scala#L124-L135) which uses `pollInterval`. However, I didn't manage to have the `timer` execute in `bindHostPortsToAddr`. I am not an expert in `Var` and `Addr` but my suspicion is that this test setup bypasses the timer:

    val addr = inetResolverWithPool.bind("127.0.0.1:80")
    val f = addr.changes.filter(_ != Addr.Pending).toFuture

I would be happy if you could suggest a different test setup, so that the timer is executed. In that case we could have a `CountDownLatch(2)` and put `Time.withCurrentTimeFrozen` to use and as a bonus bump test coverage :)